src/main/java/frc/team2412/robot/util/SDSwerveModule.java

```diff
@@ -0,0 +1,100 @@
+package frc.team2412.robot.util;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.swervedrivespecialties.swervelib.Mk4ModuleConfiguration;
+import com.swervedrivespecialties.swervelib.ModuleConfiguration;
+import com.swervedrivespecialties.swervelib.SdsModuleConfigurations;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+public class SDSwerveModule {
+
+    private static final double ENCODER_TICKS_PER_ROTATION = 2048;
+    private static final ModuleConfiguration moduleType = SdsModuleConfigurations.MK4_L1;
+    private static final Mk4ModuleConfiguration moduleConfigs = new Mk4ModuleConfiguration();
+
+    private static final double DRIVE_MOTOR_ENCODER_TICKS_TO_METERS = 1 / ENCODER_TICKS_PER_ROTATION
+            * moduleType.getDriveReduction() * Math.PI * moduleType.getWheelDiameter();
+    private static final double DRIVE_MOTOR_ENCODER_VELOCITY_TO_METERS_PER_SECOND = DRIVE_MOTOR_ENCODER_TICKS_TO_METERS
+            * 10.0;
+
+    private static final double TURN_MOTOR_ENCODER_TICKS_TO_DEGREE = 1 / ENCODER_TICKS_PER_ROTATION
+            * moduleType.getSteerReduction() * 360;
+
+    private static final double DRIVE_MOTOR_P = 1;
+    private static final double DRIVE_MOTOR_D = 0.1;
+
+    private static final double TURN_MOTOR_P = 0.2;
+    private static final double TURN_MOTOR_D = 0.1;
+
+    private final WPI_TalonFX driveMotor;
+    private final WPI_TalonFX turnMotor;
+
+    public SDSwerveModule(int driveMotorPort, int turnMotorPort, String canbus) {
+        driveMotor = new WPI_TalonFX(driveMotorPort, canbus);
+        turnMotor = new WPI_TalonFX(turnMotorPort, canbus);
+
+        TalonFXConfiguration driveMotorSettings = new TalonFXConfiguration();
```

Comment:
The drive motor isn't actually configured with these settings (and the turn motor isn't configured either)...